src/com/zone5cloud/http/core/api/UserAPI.java

```diff
@@ -107,6 +113,7 @@ public void onError(int code, Z5Error error) {
 				if (handler != null) {
 					handler.onError(code, error);
 				}
+				client.setUserName(null);
```

Comment:
not in onError(). in logout() - look at line 140 where the logout callback calls `getClient().setToken(null);` This is where you want to also call client.setUserName(null);